### PR TITLE
Fix compilation with Gtest 1.8.0 and GCC 5.4.1

### PR DESCRIPTION
**Pull request by heylouiz**
Hi,
As I said in #537 I'm sending this PR that fix the compilation with Gtest 1.8.0 and GCC 5.4.1.

### CODE REVIEW

**Comment by redboltz** [comment]
Thank you for sending the PR. I commented in the code.

File: test/carray.cpp [hunk 1]
```diff
@@ -113,7 +113,7 @@ TEST(carray, pack_unpack_char_as_str)
     msgpack::pack(ss, v1);
     msgpack::object_handle oh = msgpack::unpack(ss.str().data(), ss.str().size());
     EXPECT_EQ(oh.get().type, msgpack::type::STR);
-    EXPECT_EQ(oh.get().via.str.size, 1);
+    EXPECT_EQ(oh.get().via.str.size, (size_t) 1);
```
Comment (redboltz) [inline test/carray.cpp]: Could you replace C-style cast like `(size_t) 1` with `static_cast<size_t>(1)`?

**Comment by heylouiz** [comment]
Done.

I think the compilation problem has nothing to do with the gtest version, it's just new warning from GCC 5.4.

**Comment by redboltz** [comment]
@heylouiz , thank you very much for updating the PR!
I merge it.

**Pull request merged**